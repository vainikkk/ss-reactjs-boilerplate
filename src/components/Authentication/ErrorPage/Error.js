import React from "react";
import { useTranslation } from "react-i18next";

import UserUnauthorized from "assets/images/403.png";

import "./Error.scss";

const Error = () => {
  const { t } = useTranslation();
  return (
    <section className="unauthorized-container">
      <div className="box-center">
        <div className="image-403">
          <img src={UserUnauthorized} alt="" />
        </div>
        <div className="content">
          <p>{t("auth.unauthorized")}</p>
        </div>
      </div>
    </section>
  );
};

export default Error;
