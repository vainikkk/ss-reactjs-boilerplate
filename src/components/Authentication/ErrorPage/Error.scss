.unauthorized-container {
  background: #fff;
  display: flex;
  height: 100%;
  width: 100%;
  align-items: center;
  justify-content: center;
  text-align: center;

  .box-center {
    .image-403 {
      font-size: 6vw;
    }

    .content {
      p {
        font-size: 26px;
      }
    }
  }
}
