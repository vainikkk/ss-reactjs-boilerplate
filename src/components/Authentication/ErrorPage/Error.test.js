import React from "react";
import { shallow } from "enzyme";
import Error from "./Error";

//  use for override i18next configuration
// jest.mock("react-i18next", () => ({
//   // this mock makes sure any components using the translate hook can use it without a warning being shown
//   useTranslation: () => {
//     return {
//       t: (str) => str,
//       i18n: {
//         changeLanguage: () => new Promise(() => {}),
//       },
//     };
//   },
// }));

describe("Error", () => {
  it("should render correctly without any error", () => {
    const component = shallow(<Error />);

    expect(component).toMatchSnapshot();
  });
});
