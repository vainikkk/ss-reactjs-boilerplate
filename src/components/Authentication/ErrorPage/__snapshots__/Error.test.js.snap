// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should render correctly without any error 1`] = `
<section
  className="unauthorized-container"
>
  <div
    className="box-center"
  >
    <div
      className="image-403"
    >
      <img
        alt=""
        src="403.png"
      />
    </div>
    <div
      className="content"
    >
      <p>
        You are not authorized to access this page. Please try again.
      </p>
    </div>
  </div>
</section>
`;
